kubectl-rook-ceph.sh

```diff
@@ -273,6 +274,90 @@ function run_purge_osd() {
+  fi
+  KUBECTL_NS_CLUSTER get pod -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName | grep mon | grep --invert-match canary
+}
+
+function check_mon_quorum() {
+  echo -e "${INFO_PREFIX}Checking mon quorum and ceph health details"
+  ceph_health_details=$(KUBECTL_NS_OPERATOR exec deploy/rook-ceph-operator -- ceph health detail --conf="$CEPH_CONF_PATH")
+  if [[ "$ceph_health_details" = "HEALTH_OK" ]]; then
+    echo -e "$ceph_health_details"
+  elif [[ "$ceph_health_details" =~ "HEALTH_WARN" ]]; then
+    echo -e "${WARNING_PREFIX}$ceph_health_details"
+  elif [[ "$ceph_health_details" =~ "HEALTH_ERR" ]]; then
+    echo -e "${ERROR_PREFIX}$ceph_health_details"
+  fi
+}
+
+function check_osd_pod_count_and_nodes() {
+  echo -e "${INFO_PREFIX}Checking if atleast three osd pods are running on different nodes"
+  osd_unique_node_count=$(KUBECTL_NS_CLUSTER get pod -o=custom-columns=NAME:.metadata.name,NODE:.spec.nodeName | grep osd | grep -v prepare | awk '{print $2}' | sort | uniq | wc -l)
+  if [ "$osd_unique_node_count" -lt 3 ]; then
+    echo -e "${WARNING_PREFIX}Atleast three osd pods should running on different node"
+  fi
+  KUBECTL_NS_CLUSTER get pod -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName | grep osd | grep --invert-match prepare
+}
+
+function check_all_pods_status() {
+  echo -e "${INFO_PREFIX}Pods that are in 'Running' status"
+  KUBECTL_NS_OPERATOR get pod --field-selector status.phase=Running
+  if [ "$ROOK_OPERATOR_NAMESPACE" != "$ROOK_CLUSTER_NAMESPACE" ]; then
+    KUBECTL_NS_CLUSTER get pod --field-selector status.phase=Running
+  fi
+
+  echo
+  echo -e "${WARNING_PREFIX}Pods that are 'Not' in 'Running' status"
+  KUBECTL_NS_OPERATOR get pod --field-selector status.phase!=Running | grep --invert-match Completed
+  if [ "$ROOK_OPERATOR_NAMESPACE" != "$ROOK_CLUSTER_NAMESPACE" ]; then
+    KUBECTL_NS_CLUSTER get pod --field-selector status.phase!=Running | grep --invert-match Completed
+  fi
+}
+
+function check_pg_are_active_clean() {
+  echo -e "${INFO_PREFIX}checking placement group status"
+  pg_state=$(KUBECTL_NS_OPERATOR exec deploy/rook-ceph-operator -- ceph pg stat --conf="$CEPH_CONF_PATH")
+  pg_state_code=$(echo "${pg_state}" | awk '{print $4}')
+  if [[ "$pg_state_code" = *"active+clean;"* ]]; then
+    echo -e "${INFO_PREFIX}$pg_state"
+  elif [[ "$pg_state_code" =~ down || "$pg_state_code" =~ incomplete || "$pg_state_code" =~ snaptrim_error ]]; then
+    echo -e "${ERROR_PREFIX}$pg_state"
+  else
+    echo -e "${WARNING_PREFIX}$pg_state"
+  fi
+}
+
+function check_mgr_pods_status_and_count() {
+  echo -e "${INFO_PREFIX}checking if at least one mgr pod is running"
+  mgr_pod_count=$(KUBECTL_NS_CLUSTER get pod -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase | grep mgr | awk '{print $2}' | sort | uniq | wc -l)
+  if [ "$mgr_pod_count" -lt 1 ]; then
+    echo -e "${ERROR_PREFIX}At least one mgr pod should running"
```

Comment:
For readability, to go with the other space suggestion:
```suggestion
    echo -e "${ERROR_PREFIX} At least one mgr pod should running"
```